_posts/2020-04-14-s3-access.md

```diff
@@ -0,0 +1,212 @@
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "0123456789", "123456790" ]
+      }
+  }
+  ```
+
+  ```json
+  {
+      "Principal": {
+         "AWS": [ "arn:aws:iam::0123456789:root", "arn:aws:iam::123456790:root" ]
+      }
+  }
+  ```
+
+  Give access to everyone in the world
+  ```json
+  {
+      "Principal": "*"
+  }
+  ```
+
+  Give access to a specific role within an account
+  ```json
+  {
+      "Principal": {
+          "AWS": "arn:aws:iam::1234567890:role/my-cool-role"
+      }
+  }
+  ```
+
+  Note that the only time wildcards work in a policy is when they are specified as the only thing in a value - "*" will work, but "arn:aws:iam::1234567890:role/*" will not.
+
+- Condition
+
+  Condition statements allow us to further restrict access based on various circumstantial attributes. Maybe we block Mark all the time, except for when Mark is accessing the bucket from a particular office.
+
+- Canonical ID
+
+  Canonical ID's pop up in a few places but you are more likely to see them in ACLs than anywhere else. They usually identify what accounts may have access to certain buckets.
+
+- Resource Policy & IAM Policies
+
+  Resource policies are access policies that apply to 'resources' as opposed to 'identities' (like IAM Policies). Resource policies are most notable in that they include a 'principal' statement to narrow down the identity accessing the resource. Identity policies do not have a principal statement because they attached to identities - the principal is therefore implied.
+
+  Note: As dicussed before, principal statements can not include wildcard ('*') statements.
+
+- Owner
+
+  All resources in S3 have a concept of an 'owner'. The owner of the bucket is always the account that created it, and so far as I know can't be changed. However, objects in that bucket may be owned by other accounts. The bucket owner pays the bills, and can therefore deny access to objects and delete objects that are owned by others in the bucket. The bucket owner only has full access to objects that they own, unless the object owner grants them full access.
+
+  The object owner can only be changed by overwriting the object - there is no way to change ownership of an existing object.
+
+  This admittedly doesn't come up too often, but it is good to be aware of it.
+
+With that out of the way, let's get a bit more practical by looking at most of the access mechanisms in S3.
+
+# What is Public Access?
+
+  A public bucket is one that grants access to it's objects without requiring any proof of identity. Because of how confusing S3 policies can be, AWS now has methods to deny public access to buckets, regardless of whether a policy will technically allow it. If you are trying to either allow or deny public access to a bucket, you should check those settings first - of which we will detail now.
```

Comment:
grammar correction on first sentence for 'its' to be:
`A public bucket is one that grants access to its objects without requiring any proof of identity.`